### PR TITLE
prov/rxm: Updates to rxm provider to suppress warnings

### LINKED ISSUE #3369: prov/rxm: build warnings on windows
Snippet from AppVeyor logs:

prov\rxd\src\rxd_init.c(38): warning C4459: declaration of 'rxd_info' hides global declaration [C:\projects\libfabric\libfabric.vcxproj]
  c:\projects\libfabric\prov\rxd\src\rxd.h(94): note: see declaration of 'rxd_info'
  rxd_rma.c
C:\projects\libfabric\include\windows/osd.h(863): warning C4244: '=': conversion from 'int' to 'float', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
  rxm.c
C:\projects\libfabric\include\windows/osd.h(863): warning C4244: '=': conversion from 'int' to 'float', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
  rxm_attr.c
C:\projects\libfabric\include\windows/osd.h(863): warning C4244: '=': conversion from 'int' to 'float', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_attr.c(44): warning C4245: 'initializing': conversion from 'int' to 'uint64_t', signed/unsigned mismatch [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_attr.c(52): warning C4245: 'initializing': conversion from 'int' to 'uint64_t', signed/unsigned mismatch [C:\projects\libfabric\libfabric.vcxproj]
  rxm_conn.c
C:\projects\libfabric\include\windows/osd.h(863): warning C4244: '=': conversion from 'int' to 'float', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_conn.c(230): warning C4389: '!=': signed/unsigned mismatch [C:\projects\libfabric\libfabric.vcxproj]
  rxm_rma.c
C:\projects\libfabric\include\windows/osd.h(863): warning C4244: '=': conversion from 'int' to 'float', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_rma.c(75): warning C4244: 'return': conversion from 'ssize_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_rma.c(173): warning C4244: 'return': conversion from 'ssize_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_rma.c(178): warning C4244: 'return': conversion from 'ssize_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_rma.c(218): warning C4244: '=': conversion from 'ssize_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
  rxm_cq.c
C:\projects\libfabric\include\windows/osd.h(863): warning C4244: '=': conversion from 'int' to 'float', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_cq.c(180): warning C4244: '=': conversion from 'int' to 'uint8_t', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_cq.c(222): warning C4244: '=': conversion from 'ssize_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_cq.c(256): warning C4244: '=': conversion from 'uint64_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_cq.c(437): warning C4244: '=': conversion from 'ssize_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_cq.c(598): warning C4018: '<': signed/unsigned mismatch [C:\projects\libfabric\libfabric.vcxproj]
  Compiling...
  rxm_domain.c
C:\projects\libfabric\include\windows/osd.h(863): warning C4244: '=': conversion from 'int' to 'float', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
  rxm_ep.c
C:\projects\libfabric\include\windows/osd.h(863): warning C4244: '=': conversion from 'int' to 'float', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_ep.c(164): warning C4244: '=': conversion from 'int' to 'uint8_t', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_ep.c(175): warning C4244: 'function': conversion from 'int' to 'uint8_t', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_ep.c(289): warning C4244: '=': conversion from 'ssize_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_ep.c(521): warning C4267: '=': conversion from 'size_t' to 'uint8_t', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_ep.c(666): warning C4267: '=': conversion from 'size_t' to 'uint8_t', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_ep.c(706): warning C4267: '=': conversion from 'size_t' to 'uint8_t', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_ep.c(717): warning C4244: '=': conversion from 'int' to 'uint8_t', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_ep.c(752): warning C4244: '=': conversion from 'ssize_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_ep.c(773): warning C4244: '=': conversion from 'ssize_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
prov\rxm\src\rxm_ep.c(787): warning C4244: '=': conversion from 'ssize_t' to 'int', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]
  rxm_fabric.c
C:\projects\libfabric\include\windows/osd.h(863): warning C4244: '=': conversion from 'int' to 'float', possible loss of data [C:\projects\libfabric\libfabric.vcxproj]

### PR DESCRIPTION
**Pull request by gladkovdmitry17**
Fixes  #3369

### CODE REVIEW

**Comment by gladkovdmitry17** [comment]
@a-ilango Could you take a look, please?

File: prov/rxm/src/rxm_cq.c [hunk 1]
```diff
@@ -414,7 +414,8 @@ static int rxm_lmt_send_ack(struct rxm_rx_buf *rx_buf)
 		return -FI_EAGAIN;
 	}
 
-	if (!(tx_entry = rxm_tx_entry_get(&rx_buf->ep->send_queue))) {
```
Comment (a-ilango) [inline prov/rxm/src/rxm_cq.c]: Was this a warning in windows? I'm fine with the change. Just curious.

Comment (gladkovdmitry17) [reply prov/rxm/src/rxm_cq.c]: unfortunately, the Windows compiler complains that we are doing an assignment in the condition.
I tried to disable this warning via `#pragma`. But in this case the code like this
```
if (a = 5) { ... }
```
looks good for compiler w/o any warnings.
I guess it's more better to restrict some minor convention for the C syntax than allow a potentially incorrect condition w/o any warnings

Comment (gladkovdmitry17) [reply prov/rxm/src/rxm_cq.c]: Just as an example for the AppVeyor CI testing:

```
c:\projects\libfabric\prov\util\src\util_poll.c(102): warning C4706: assignment within conditional expression
```

Comment (a-ilango) [reply prov/rxm/src/rxm_cq.c]: Thanks for the clarification!

**Pull request merged**